src/client/dfuse/ops/statfs.c

```diff
@@ -0,0 +1,68 @@
+/**
+ * (C) Copyright 2020 Intel Corporation.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * GOVERNMENT LICENSE RIGHTS-OPEN SOURCE SOFTWARE
+ * The Government's rights to use, modify, reproduce, release, perform, display,
+ * or disclose this software are subject to the terms of the Apache License as
+ * provided in Contract No. B609815.
+ * Any reproduction of computer software, computer software documentation, or
+ * portions thereof marked with this legend must also reproduce the markings.
+ */
+
+#include "dfuse_common.h"
+#include "dfuse.h"
+
+void
+dfuse_cb_statfs(fuse_req_t req, struct dfuse_inode_entry *inode)
+{
+	struct statvfs stbuf = {};
+	daos_pool_info_t info = {.pi_bits = DPI_SPACE};
+	int rc;
+
+	if (!daos_handle_is_inval(inode->ie_dfs->dfs_poh)) {
+		rc = daos_pool_query(inode->ie_dfs->dfs_poh, NULL, &info,
+				     NULL, NULL);
+		if (rc != -DER_SUCCESS)
+			D_GOTO(err, rc = daos_der2errno(rc));
+
+		stbuf.f_blocks = info.pi_space.ps_space.s_total[DAOS_MEDIA_SCM] \
+			+ info.pi_space.ps_space.s_total[DAOS_MEDIA_NVME];
+		stbuf.f_bfree = info.pi_space.ps_space.s_free[DAOS_MEDIA_SCM] \
+			+ info.pi_space.ps_space.s_free[DAOS_MEDIA_NVME];
+
+		DFUSE_TRA_INFO(inode, "blocks %#lx free %#lx",
+			       stbuf.f_blocks, stbuf.f_bfree);
+	} else {
+		stbuf.f_blocks = 1024 * 1024 * 2;
+		stbuf.f_bfree = 1024 * 1024;
```

Comment:
shouldn't we return an error here in this case? (when the pool handle is invalid)

Reply:
This is the case we fall into when you launch dfuse without a pool option and run "df", if we return valid values (specifically non-zero f_blocks) then dfuse shows up in the df output, if we don't then dfuse will continue to be invisible to df.  If you start dfuse with a pool on the command line, or call df and specify a path below then mount point then it falls into the other case and returns the pool size correctly.

Reply:
well IMO it would be better for it not to show up rather than to return those values. just my opinion so it would be good to get others opinions about his.

Reply:
One of the frequently asked questions we get in dfuse presentations has been "why does dfuse not show up in df, and how do we tell it's running if it doesn't".  I'm not a fan of using hard-coded values here but it does solve a problem, and the inode values are inherently made up anyway.

Reply:
-1 works really well here, new version pushed.